packages/@o3r/storybook/schematics/ng-add/index.ts

```diff
@@ -1,20 +1,33 @@
-import { chain, noop, Rule } from '@angular-devkit/schematics';
-import { applyEsLintFix, install } from '@o3r/schematics';
+import { chain, noop, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import * as path from 'node:path';
 import { NgAddSchematicsSchema } from './schema';
-import { updateStorybook } from '../storybook-base';
-import { isPackageInstalled } from '@o3r/dev-tools';
-
 
 /**
  * Add Otter storybook to an Angular Project
  *
  * @param options
  */
 export function ngAdd(options: NgAddSchematicsSchema): Rule {
-  return chain([
-    // if the package is already installed skip the updates on files to avoid overriding the existing configurations
-    isPackageInstalled('@otter/storybook') || isPackageInstalled('@o3r/storybook') ? noop() : updateStorybook(options, __dirname),
-    options.skipLinter ? noop() : applyEsLintFix(),
-    options.skipInstall ? noop() : install
-  ]);
+  return async (_tree: Tree, context: SchematicContext) => {
+    try {
+      const { applyEsLintFix, install, ngAddPackages, getPeerDepWithPattern } = await import('@o3r/schematics');
+      const { updateStorybook } = await import('../storybook-base');
```

Comment:
not sure it needs to be dynamic import

Reply:
is it for this one?

Reply:
yes, the import of o3r/schematics is inside '...storybook-base/index.ts'